apps/main/lib/main/views/home.rb

```diff
@@ -0,0 +1,18 @@
+require "main/import"
+require "main/view/controller"
+
+module Main
+  module Views
+    class Home < Main::View::Controller
+      include Main::Import["repositories.articles_repo"]
+
+      configure do |config|
+        config.template = "home"
+      end
+
+      expose :articles do
+        articles_repo.home_page_listing
```

Comment:
And here's a few more notes about your questions, @choonkeat:

> > does "repositories.articles_repo" do a require_relative "../../repositories/articles_repo.rb"?
> No, it does require "main/repositories/articles_repo" because we have load paths set

It's probably worth noting that [this](https://github.com/dry-rb/dry-web-blog/blob/741609eedc59599b52e29c5aa77a30073a95721e/apps/main/system/boot.rb#L3) is the part of our app that triggers this `require` to happen. When we run `Main::Container.finalize!`, it crawls over the container's auto_register paths ([configured here](https://github.com/dry-rb/dry-web-blog/blob/741609eedc59599b52e29c5aa77a30073a95721e/apps/main/system/main/container.rb#L15)) and for each Ruby source file it finds, it `require`s that file, then makes a matching container registration that'll return an instance of the class defined in that file. So by the time this `"repositories.articles_repo"` dependency is injected here, it's already had its file required.

(FWIW, there's a different story when it comes to unit testing – in those cases, we don't finalize the container, and it will lazily load/require files for us at the first moment we try and resolve them from the container, which makes for a much faster test cycle, since it doesn't load unnecessary files.)

> variable articles_repo name is extracted from the string

Yeah, the name for the local getter can be inferred from the dependency identifier like you've seen here, but we can also be explicit in naming them:

```ruby
inclue Main::Import[repo: "persistence.articles_repo"]
```

This would give us `repo` (instead of `articles_repo`) as the local name for the dependency.